docs/programming_guide.rst

```diff
@@ -0,0 +1,83 @@
+
+The following gives a short overview over the chapters of the HIP programming
+guide, in order to guide you through the programming guide and explain how to
+make the best use of HIP.
+
+Write GPU Kernels for Parallel Execution
+================================================================================
+
+To make the most of the parallelism inherent to GPUs, a thorough understanding
+of the :ref:`programming model <programming_model>` is helpful. HIPs'
+programming model is designed to make it easy to map data-parallel algorithms to
+architecture of the GPUs. HIP employs the so-called SIMT-model (Single
+Instruction Multiple Threads) with a multi-layered thread hierarchy for
+efficient execution.
+
+Understand the Target Architecture (CPU + GPU)
+================================================================================
+
+The general :ref:`hardware implementation <hardware_implementation>` of GPUs
+supported by HIP is outlined in this chapter. In general, GPUs are made up of
+many so called Compute Units that excel at executing parallelizable,
+computationally intensive workloads without complex control-flow.
+
+Increase parallelism on multiple level
+================================================================================
+
+To maximize performance and keep all system components fully utilized, the
+application should expose and efficiently manage as much parallelism as possible.
+:ref:`Parallel execution <parallel execution>` can be achieved at the
+application, device, and multiprocessor levels.
+
+The application’s host and device operations can achieve parallel execution
+through asynchronous calls, streams, or HIP graphs. On the device level,
+multiple kernels can execute concurrently when resources are available, and at
+the multiprocessor level, developers can overlap data transfers with
+computations to further optimize performance.
+
+Memory management
+================================================================================
+
+GPUs generally have their own destinct memory, also referred to as :ref:`device
+memory <device_memory>`, separate from the :ref:`host memory <device_memory>`.
+This memory needs to be managed separately from the host memory. This includes
+allocating the memory and transfering it between the host and the device. These
+operations can be performance critical, so it is important to know how to use
+them effectively.
+
+For more information, see :ref:`memory management <memory_management>`.
+
+Synchronize CPU and GPU Workloads
+================================================================================
+
+Tasks on the host and devices run asynchronously, so proper synchronization is
+needed when dependencies between those tasks exist. Asynchronous execution of
+tasks is useful to fully utilize the available resources. Even when only a
+single device is available, memory transfers and the execution of tasks can be
+overlapped with asynchrononus execution.
```

Comment:
```suggestion
overlapped with asynchronous execution. For more information, see XXX. 
```

Reply:
This topic is not written yet. We will link in, when in the PR with the new section.